clients/go/wire.go

```diff
@@ -0,0 +1,67 @@
+package raccoon
+
+import (
+	"errors"
+
+	"encoding/json"
+
+	"google.golang.org/protobuf/proto"
+)
+
+// WireMarshaler defines a conversion between byte sequence and raccoon request payloads.
+type WireMarshaler interface {
+
+	// Marshal marshals "any" into byte sequence.
+	Marshal(any interface{}) ([]byte, error)
```

Comment:
In a way the functions defined in this file should be an extensions of the interface in marshal.go?

Reply:
No, they appear similar, but they serve different purposes, so they are kept separate.

Reply:
Yes, they are different. One more reason IMO to call the external one as serializer.

Reply:
Fine.